sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -155,7 +155,12 @@ object SQLConf {
           confGetter.get()()
         }
       } else {
-        confGetter.get()()
+        val conf = existingConf.get()
+        if (conf != null) {
```

Comment:
How could this happen?

Reply:
It happens in tests, but in general I think if you put an override in place it should always work. It was kind of unexpected that the test didn't pick up the SQLConf I was setting.

Reply:
seems like this is a new feature for testing: now we can overwrite the current SQLConf by calling `SQLConf.withExistingConf`.

This might be a good idea, but this is not how we handle this case in the current codebase. If you look at `ParquetFilterSuite`, we can set the current SQLConf in `beforeAll`, and unset it in `afterAll`. How about we follow the existing solution and open a new PR for this new feature?